pyomo/gdp/plugins/bigm.py

```diff
@@ -792,113 +799,26 @@ def _estimate_M(self, expr, name):
-                    "transformed" % constraint.name)
-            parent_disjunct = parent_disjunct.parent_block()
-
-        return parent_disjunct
-
-    def _get_constraint_transBlock(self, constraint):
-        parent_disjunct = self._find_parent_disjunct(constraint)
-        # we know from _find_parent_disjunct that parent_disjunct is a Disjunct,
-        # so the below is OK
-        transBlock = parent_disjunct._transformation_block
-        if transBlock is None:
-            raise GDP_Error("Constraint %s is on a disjunct which has not been "
-                            "transformed" % constraint.name)
-        # if it's not None, it's the weakref we wanted.
-        transBlock = transBlock()
-
-        return transBlock
-
-    def get_transformed_constraint(self, srcConstraint):
-        """Return the transformed version of srcConstraint
-
-        Parameters
-        ----------
-        srcConstraint: Constraint, which must be in the subtree of a
-                       transformed Disjunct
-        """
-        transBlock = self._get_constraint_transBlock(srcConstraint)
-        
-        if hasattr(transBlock, "_constraintMap") and transBlock._constraintMap[
-                'transformedConstraints'].get(srcConstraint):
-            return transBlock._constraintMap['transformedConstraints'][
-                srcConstraint]
-        raise GDP_Error("Constraint %s has not been transformed." 
-                        % srcConstraint.name)
+    def get_transformed_constraints(self, srcConstraint):
+        return get_transformed_constraints(srcConstraint)
 
     def get_src_disjunction(self, xor_constraint):
-        """Return the Disjunction corresponding to xor_constraint
-
-        Parameters
-        ----------
-        xor_constraint: Constraint, which must be the logical constraint 
-                        (located on the transformation block) of some 
-                        Disjunction
-        """
-        # NOTE: This is indeed a linear search through the Disjunctions on the
-        # model. I am leaving it this way on the assumption that asking XOR
-        # constraints for their Disjunction is not going to be a common
-        # question. If we ever need efficiency then we should store a reverse
-        # map from the XOR constraint to the Disjunction on the transformation
-        # block while we do the transformation. And then this method could query
-        # that map.
-        m = xor_constraint.model()
-        for disjunction in m.component_data_objects(Disjunction):
-            if disjunction._algebraic_constraint:
-                if disjunction._algebraic_constraint() is xor_constraint:
-                    return disjunction
-        raise GDP_Error("It appears that %s is not an XOR or OR constraint "
-                        "resulting from transforming a Disjunction."
-                        % xor_constraint.name)
+        return get_src_disjunction(xor_constraint)
```

Comment:
I agree with the code reduction that comes with aggregating everything into `gdp.util`, but I am a bit sad that we are losing the nice docstrings. I think there ought to be a way to do something similar to `functool.wraps` in this case.

Maybe something to consider after merge. I wouldn't hold this up on that account.

Reply:
Fixed.  I made use of `@functools.wraps` so that the transformation methods inherif the docstring from the implementations in utils.